src/orion/algo/robo/base.py

```diff
@@ -181,129 +222,115 @@ class RoBO(BaseAlgorithm):
-    requires_dist = "linear"
-    requires_shape = "flattened"
+    requires_type: ClassVar[str] = "real"
+    requires_dist: ClassVar[str] = "linear"
+    requires_shape: ClassVar[str] = "flattened"
 
     def __init__(
         self,
-        space,
-        seed=0,
-        n_initial_points=20,
-        maximizer="random",
-        acquisition_func="log_ei",
-        **kwargs,
+        space: Space,
+        seed: int | Sequence[int] | None = 0,
+        n_initial_points: int = 20,
+        maximizer: MaximizerName = "random",
+        acquisition_func: AcquisitionFnName = "log_ei",
     ):
-
-        self.model = None
-        self.robo = None
-        self._bo_duplicates = []
-
-        super(RoBO, self).__init__(
-            space,
-            n_initial_points=n_initial_points,
-            maximizer=maximizer,
-            acquisition_func=acquisition_func,
-            seed=seed,
-        )
-
-        # Otherwise it is turned no 'random' because of BaseAlgorithms constructor... -_-
+        super().__init__(space)
+        self._bo_duplicates: dict[str, Trial] = {}
+        self.n_initial_points = n_initial_points
+        self.maximizer: MaximizerName = maximizer
+        self.acquisition_func: AcquisitionFnName = acquisition_func
+        self.seed = seed
         self.maximizer = maximizer
 
-        self._param_names += list(kwargs.keys())
-        for key, value in kwargs.items():
-            setattr(self, key, value)
-
-    @property
-    def space(self):
-        """Space of the optimizer"""
-        return self._space
-
-    @space.setter
-    def space(self, space):
-        """Setter of optimizer's space.
-
-        Side-effect: Will initialize optimizer.
-        """
-        self._original = self._space
-        self._space = space
-        self._initialize()
-
-    def _initialize_model(self):
-        """Build model and register it as ``self.model``"""
-        raise NotImplementedError()
-
-    def build_acquisition_func(self):
-        """Build and return the acquisition function."""
-        return build_acquisition_func(self.acquisition_func, self.model)
-
-    def _initialize(self):
-        """Initialize the optimizer once the space is transformed"""
-        self._initialize_model()
+        # "abstract" attributes, set in `_initialize`
+        self.model: ModelType
+        self.robo: BayesianOptimization
+        self.rng: numpy.random.RandomState
+        self._initialized: bool = False
+
+        # TODO: Remove once the fix of https://github.com/Epistimio/orion/pull/947 is included in
+        # a release of orion.
+        init_signature = inspect.signature(type(self).__init__)
+        self._param_names = [
+            name
+            for name, param in init_signature.parameters.items()
+            if name not in ["self", "space"]
+            and param.kind not in [param.VAR_KEYWORD, param.VAR_POSITIONAL]
+        ]
+
+    def _initialize(self, seed_rng: bool = True) -> None:
```

Comment:
Could this be part of `__init__` now that we don't need to wait for the transformed space to be set?

Reply:
Not quite, this isn't to do with the space, it's to do with the creation of the model and optimizer and the seeding.

I'll double-check if It's possible to remove this, but if it isn't, I'll rename it to something more appropriate.

Reply:
I just realized why this is here.

The problem is, build_model needs to be called eventually to create the model, and the implementations in subclasses depend on some addiitonal attributes that are set on self after super().init in the subclasses. Therefore, it can't be called by the base class's init, but it has to be called at some point before we can suggest a new point. Hence the two-stage initialization.

Could this stay as-is? Or would you have any ideas on how this could be cleaner?

Reply:
Ok that's good. I just wanted to make sure it wasn't just because of space setter anymore.